Sanity/per-connection-ssl/runtest.sh

```diff
@@ -0,0 +1,260 @@
+      )
+EOF
+
+    rsyslogConfigAddTo "RULES" < <(rsyslogConfigCreateSection 'SSL')
+    rsyslogServerConfigAddTo "MODULES" < <(rsyslogConfigCreateSection 'MODL')
+    rsyslogServerConfigAddTo "RULES" < <(rsyslogConfigCreateSection 'RULESETS')
+    rsyslogServerConfigAddTo "RULES" < <(rsyslogConfigCreateSection 'CONN1')
+    rsyslogServerConfigAddTo "RULES" < <(rsyslogConfigCreateSection 'CONN2')
+    rsyslogServerConfigAddTo "RULES" < <(rsyslogConfigCreateSection 'CONN3')
+
+    rsyslogConfigReplace "SSL" << EOF
+      local1.* action(
+        type="omfwd"
+        Protocol="tcp"
+        Target="127.0.0.1"
+        Port="50514"
+        StreamDriver="$driver"
+        StreamDriverMode="1"
+        StreamDriverAuthMode="x509/certvalid"
+      )
+
+      local2.* action(
+        type="omfwd"
+        Protocol="tcp"
+        Target="127.0.0.1"
+        Port="50515"
+        StreamDriver="$driver"
+        StreamDriverMode="1"
+        StreamDriverAuthMode="x509/certvalid"
+        streamDriver.CAFile="/etc/rsyslog.d/ca1.pem"
+        streamDriver.CertFile="/etc/rsyslog.d/client1-cert.pem"
+        streamDriver.KeyFile="/etc/rsyslog.d/client1-key.pem"
+      )
+
+      local3.* action(
+        type="omfwd"
+        Protocol="tcp"
+        Target="127.0.0.1"
+        Port="50516"
+        StreamDriver="$driver"
+        StreamDriverMode="1"
+        StreamDriverAuthMode="x509/certvalid"
+        streamDriver.CAFile="/etc/rsyslog.d/ca2.pem"
+        streamDriver.CertFile="/etc/rsyslog.d/client2-cert.pem"
+        streamDriver.KeyFile="/etc/rsyslog.d/client2-key.pem"
+      )
+
+      local4.* action(
+        type="omfwd"
+        Protocol="tcp"
+        Target="127.0.0.1"
+        Port="50515"
+        StreamDriver="$driver"
+        StreamDriverMode="1"
+        StreamDriverAuthMode="x509/certvalid"
+        streamDriver.CAFile="/etc/rsyslog.d/ca2.pem"
+        streamDriver.CertFile="/etc/rsyslog.d/client2-cert.pem"
+        streamDriver.KeyFile="/etc/rsyslog.d/client2-key.pem"
+      )
+EOF
+
+    rsyslogServerConfigReplace "MODL" << EOF
+      module(
+        load="imtcp"
+        StreamDriver.AuthMode="x509/certvalid"
+        StreamDriver.Mode="1"
+        StreamDriver.Name="$driver"
+      )
+EOF
+
+    rsyslogServerConfigReplace "RULESETS" << EOF
+      ruleset(name="TestRuleSet1"){
+        *.*     $rsyslogServerLogDir/in1
+      }
+      ruleset(name="TestRuleSet2"){
+        *.*     $rsyslogServerLogDir/in2
+      }
+      ruleset(name="TestRuleSet3"){
+        *.*     $rsyslogServerLogDir/in3
+      }
+EOF
+
+    rsyslogServerConfigReplace "CONN1" << EOF
+      input(
+        type="imtcp"
+        Port="50514"
+        ruleset="TestRuleSet1"
+      )
+EOF
+
+    rsyslogServerConfigReplace "CONN2" << EOF
+      input(
+        type="imtcp"
+        Port="50515"
+        streamdriver.CAFile="/etc/rsyslog.d/ca1.pem"
+        streamdriver.CertFile="/etc/rsyslog.d/client1-cert.pem"
```

Comment:
Inputs(servers) are using the client certificates. I think this might have been a copy-paste mistake. Am I missing something?

Reply:
Good catch.
It should not make any difference as there's the cert validity check enabled, though.
Unfortunately, it does not make the _ossl_ one to pass, neither.